ciao-controller/internal/datastore/sqlite3db.go

```diff
@@ -2304,9 +2307,26 @@ func (ds *sqliteDB) deleteBlockData(ID string) error {
 }
 
 func (ds *sqliteDB) createStorageAttachment(a types.StorageAttachment) error {
-	ds.dbLock.Lock()
-	err := ds.create("attachments", a.ID, a.InstanceID, a.BlockID)
-	ds.dbLock.Unlock()
+	datastore := ds.getTableDB("attachments")
+
+	ds.tdbLock.Lock()
+
```

Comment:
I filed an issue for the badly designed function.
https://github.com/01org/ciao/issues/775